From 56625642872e59f44d669d5e5da218a8ebbafe02 Mon Sep 17 00:00:00 2001
From: "M, Kumar K" <kumar.k.m@intel.com>
Date: Wed, 18 Sep 2019 12:16:57 +0530
Subject: [PATCH] Handle HDMI audio device in framework

Handle HDMI device in wired acc manager and default policy.

Tracked-On: OAM-84728
Signed-off-by: M, Kumar K <kumar.k.m@intel.com>
---
 .../com/android/server/WiredAccessoryManager.java    | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)
 mode change 100644 => 100755 services/core/java/com/android/server/WiredAccessoryManager.java

diff --git a/services/core/java/com/android/server/WiredAccessoryManager.java b/services/core/java/com/android/server/WiredAccessoryManager.java
old mode 100644
new mode 100755
index 8e5c73bfc02..7d2c5cd426b
--- a/services/core/java/com/android/server/WiredAccessoryManager.java
+++ b/services/core/java/com/android/server/WiredAccessoryManager.java
@@ -166,6 +166,10 @@ final class WiredAccessoryManager implements WiredAccessoryCallbacks {
                 case SW_MICROPHONE_INSERT_BIT:
                     headset = BIT_HEADSET;
                     break;
+ 
+                case SW_HEADPHONE_INSERT_BIT | SW_LINEOUT_INSERT_BIT:
+		    headset = BIT_HEADSET;
+                    break;
 
                 default:
                     headset = 0;
@@ -404,13 +408,13 @@ final class WiredAccessoryManager implements WiredAccessoryCallbacks {
             //
             // If the kernel does not have an "hdmi_audio" switch, just fall back on the older
             // "hdmi" switch instead.
-            uei = new UEventInfo(NAME_HDMI_AUDIO, BIT_HDMI_AUDIO, 0, 0);
+	    uei = new UEventInfo(NAME_HDMI_AUDIO, BIT_HDMI_AUDIO, BIT_LINEOUT, 0);
             if (uei.checkSwitchExists()) {
                 retVal.add(uei);
             } else {
-                uei = new UEventInfo(NAME_HDMI, BIT_HDMI_AUDIO, 0, 0);
-                if (uei.checkSwitchExists()) {
-                    retVal.add(uei);
+                uei = new UEventInfo(NAME_HDMI, BIT_HDMI_AUDIO, BIT_LINEOUT, 0);
+	            if (uei.checkSwitchExists()) {
+                   retVal.add(uei);
                 } else {
                     Slog.w(TAG, "This kernel does not have HDMI audio support");
                 }
-- 
2.17.1

